pydrawise/client.py

```diff
@@ -370,3 +356,101 @@ async def get_watering_report(
                 entries,
             )
         )
+
+    async def get_watering_report(
+        self, controller: Controller, start: datetime, end: datetime
+    ) -> list[WateringReportEntry]:
+        """Retrieves a watering report for the given controller and time period.
+
+        :param controller: The controller whose watering report to generate.
+        :param start: Start time.
+        :param end: End time."""
+        selector = self._schema.Query.controller(controllerId=controller.id).select(
+            self._schema.Controller.reports.select(
+                self._schema.Reports.watering(
+                    **{
+                        "from": int(start.timestamp()),
+                        "until": int(end.timestamp()),
+                    }
+                ).select(*get_selectors(self._schema, WateringReportEntry)),
+            ),
+        )
+        result = await self._query(selector)
+        return Hydrawise._prune_watering_report_entries(
+            deserialize(
+                list[WateringReportEntry], result["controller"]["reports"]["watering"]
+            ),
+            start,
+            end,
+        )
+
+    async def get_water_use_summary(
+        self, controller: Controller, start: datetime, end: datetime
+    ) -> ControllerWaterUseSummary:
+        """Calculate the water use for the given controller and time period.
+
+        :param controller: The controller whose water use to report.
+        :param start: Start time
+        :param end: End time."""
+        selector = self._schema.Query.controller(controllerId=controller.id).select(
+            self._schema.Controller.sensors.select(
+                *get_selectors(self._schema, Sensor),
+                self._schema.Sensor.flowSummary(
+                    start=int(start.timestamp()),
+                    end=int(end.timestamp()),
+                ).select(*get_selectors(self._schema, SensorFlowSummary)),
+            ),
+            self._schema.Controller.reports.select(
+                self._schema.Reports.watering(
+                    **{
```

Comment:
Again here: no need for dict expansion.

Reply:
This doesn't work because `from` is a reserved keyword in Python.